### PR TITLE
Add instalation option over compiler sysroot

### PR DESCRIPTION
**Pull request by martinribelotta**
Actually this patch is less than ideal because meson build system not provide any form for change `prefix` into script.

instead of this, the way to install picolibc in compiler sysroot is configure the build process using new scripts `do-${arch}-configure-sysroot` (for now, only arm and riscv64)

Signed-off-by: Martin Ribelotta <martinribelotta@gmail.com>

### CODE REVIEW

File: do-arm-configure-sysroot [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+ARCH=arm-none-eabi
+DIR=`dirname $0`
+meson "$DIR" \
```
Comment (keith-packard) [inline do-arm-configure-sysroot]: Did you mean to add the -Dsysroot=true option here?

File: meson.build [hunk 2]
```diff
@@ -90,6 +90,7 @@ newlib_initfini = get_option('newlib-initfini')
 thread_local_storage = get_option('thread-local-storage')
 tls_model = get_option('tls-model')
 posix_io = get_option('posix-io')
+sysroot_install = get_option('sysroot-install')
```
Comment (keith-packard) [inline meson.build]: I don't see any uses of this new option; it sounds like it's not necessary?

Comment (martinribelotta) [reply meson.build]: This option select install_prefix or "%R" prefix in specfile. This is necesary for non absolute PATH work,

File: do-arm-configure-sysroot (hunk 1 above)
Comment (martinribelotta) [reply do-arm-configure-sysroot]: Yes! I fix it!

**Comment by keith-packard** [comment]
I tested your code by adding --sysroot= in the do-riscv-configure-sysroot script (as my compiler is built without sysroot) and it worked great! Thanks for your contribution.

**Pull request merged**